src/code/irqmgr.c

```diff
@@ -1,122 +1,174 @@
+/**
+ * @file irqmgr.c
+ *
+ * This file implements a manager for forwarding three key system interrupt events to
+ * registered clients. These interrupts are:
+ *
+ *  - VI Retrace
+ *      This event is sent to the IRQ manager by the OS VI manager which only supports
+ *      the forwarding of VI events to a single message queue. The IRQ manager will
+ *      forward these events to every registered client. VI retrace events are received
+ *      when the Video Interface has reached the start of the vertical blanking interval,
+ *      happening at approximately 60Hz on NTSC and 50Hz on PAL. Many threads sit idle
+ *      until a VI Retrace event wakes them up, at which point they will perform their
+ *      task and then return to idle to await the next retrace.
+ *
+ *  - Pre-NMI
+ *      This event is sent to the IRQ manager by the OS Interrupt Handler when the reset
+ *      button on the N64 control deck is pressed. This event is forwarded to clients to
+ *      inform them that a reset will occur in 0.5s / 500ms and to begin any shutdown
+ *      procedures.
+ *
+ *  - NMI
+ *      This event is sent at 450ms into the Pre-NMI phase, informing clients that a
+ *      reset is imminent.
+ */
 #include "global.h"
 #include "vt.h"
 
-vu32 gIrqMgrResetStatus = 0;
+vu32 gIrqMgrResetStatus = IRQ_RESET_IDLE;
 volatile OSTime sIrqMgrResetTime = 0;
 volatile OSTime gIrqMgrRetraceTime = 0;
 u32 sIrqMgrRetraceCount = 0;
 
+// Internal messages
 #define RETRACE_MSG 666
 #define PRE_NMI_MSG 669
 #define PRENMI450_MSG 671
 #define PRENMI480_MSG 672
 #define PRENMI500_MSG 673
```

Comment:
To prevent confusion with `OS_SC_NMI_MSG` and the like, should these get some kind of prefix?

Also I know you want to avoid deviating from original headers but `OS_SC_` for the messages sent by the irq with the "scheduler" being something else, is confusing

Also the `OS_SC_` defines can be used in `Sched_ThreadEntry`

Reply:
Yeah the terminology is a bit messy right now, I'm going to document sched.c next, maybe the `OS_SC_` defines should be revisited in that when both systems are in view. I also plan to add the defines to sched.c in that, but I'll update the irqmgr-internal ones now.